pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -383,6 +387,68 @@ func ensureKubeAPIServerTrustedCA(client coreclientv1.CoreV1Interface, recorder
 	return err
 }
 
+func ensureLocalhostRecoverySAToken(client coreclientv1.CoreV1Interface, recorder events.Recorder) error {
+	requiredSA := resourceread.ReadServiceAccountV1OrDie(v410_00_assets.MustAsset("v4.1.0/kube-apiserver/localhost-recovery-sa.yaml"))
+	requiredToken := resourceread.ReadSecretV1OrDie(v410_00_assets.MustAsset("v4.1.0/kube-apiserver/localhost-recovery-token.yaml"))
+
+	saClient := client.ServiceAccounts(operatorclient.TargetNamespace)
+	serviceAccount, err := saClient.Get(requiredSA.Name, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		serviceAccount, err = saClient.Create(requiredSA)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create custom secret to be populated with SA token so we have a stable name
+	secretsClient := client.Secrets(operatorclient.TargetNamespace)
+	token, err := secretsClient.Get(requiredToken.Name, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		token, err = secretsClient.Create(requiredToken)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Token creation / injection for a SA is asynchronous.
+	// We will report and error if it's missing, go degraded and get re-queued when the SA token is updated.
+
+	uid := token.Annotations[corev1.ServiceAccountUIDKey]
+	if len(uid) == 0 {
+		return fmt.Errorf("secret %s/%s hasn't been populated with SA token yet: missing SA UID", token.Namespace, token.Name)
+	}
+
+	if uid != string(serviceAccount.UID) {
+		return fmt.Errorf("secret %s/%s hasn't been populated with current SA token yet: SA UID mismatch", token.Namespace, token.Name)
+	}
+
+	if len(token.Data) == 0 {
+		return fmt.Errorf("secret %s/%s hasn't been populated with current SA data yet", token.Namespace, token.Name)
+	}
```

Comment:
unnecessary check . The if below does the same.

Reply:
checking if map is empty or if map has key "token" are not equivalent checks, but looks like I forgot to reword the error message when copy pasting

Reply:
fixed the error message

Reply:
(also this avoids indexing nil map)